magnum_cluster_api/resources.py

```diff
@@ -47,6 +49,99 @@
 PLACEHOLDER = "PLACEHOLDER"
 
 
+class BaseCli:
+    APPLY_CMD = []
+    DELETE_CMD = []
+
+    def __init__(self) -> None:
+        pass
+
+    def apply(self) -> None:
+        subprocess.run(self.APPLY_CMD, capture_output=True, check=True)
+
+    def delete(self) -> None:
+        try:
+            subprocess.run(self.DELETE_CMD, capture_output=True, check=False)
+        except subprocess.CalledProcessError as e:
+            if "release: not found" in e.output:
+                return
+            raise e
+
+
+class HelmRepository(BaseCli):
+    APPLY_CMD = "helm repo add".split()
+    DELETE_CMD = "helm repo delete".split()
+    REPO_NANME = ""
+    REPO_SOURCE = ""
+
+    def __init__(self) -> None:
+        assert self.REPO_NANME
+        assert self.REPO_SOURCE
+        self.APPLY_CMD += [self.REPO_NANME, self.REPO_SOURCE]
+        self.DELETE_CMD += [self.REPO_NANME]
+
+
+class HelmRelease(BaseCli):
+    APPLY_CMD = "helm upgrade --install".split()
+    DELETE_CMD = "helm delete".split()
+    CHART_SOURCE = ""
+    RELEASE_NAME = ""
+
+    RELEASE_NAMESPACE = ""
+    EXTRA_ARGS = []
+    VALUES = {}
+
+    def __init__(self) -> None:
+        assert self.RELEASE_NAME
+        assert self.CHART_SOURCE
+        self.APPLY_CMD += [
```

Comment:
also, in order for upgrade to be idempotent, we should use the following:

`helm upgrade --install --wait`

that way, if it already exists for some reason, it will just override.  also, it will make be helpful to wait and make sure it successfully goes up.